modules/core/src/main/java/org/apache/ignite/internal/processors/metric/MetricRegistry.java

```diff
@@ -115,12 +116,17 @@ public void reset() {
     }
 
     /**
-     * Register existing metrics in this group with the specified name.
+     * Register existing metrics in this group with the specified name. Note that the name of the metric must
+     * start with the name of the current registry it is registered into.
      *
      * @param metric Metric.
      */
     public void register(Metric metric) {
-        addMetric(metric.name(), metric);
+        String mregPrefix = regName + SEPARATOR;
+
+        assert metric.name().startsWith(mregPrefix);
+
+        addMetric(metric.name().substring(mregPrefix.length()), metric);
```

Comment:
Why we change the way we register metrics?

Reply:
To make MetricRegistry#findMetric usage consistent.

If we keep previous approach intact, metrics registered through MetricRegistry#register are found only by full metric name but this is not the case for metrics registered with other approaches - they are found by name without registry prefix .

Reply:
I think we should guarantee correct creation of the metric somehow.
Please, see my comment below - https://github.com/apache/ignite/pull/8937/files#r604746091

PS. assert should stay to check "hack" of the factory method with the `IntMetric.create("", "MyMetric", "desc");`